taca_ngi_pipeline/deliver/deliver_grus.py

```diff
@@ -290,6 +303,28 @@ def deliver_project(self):
             logger.warning('Not all the samples have been hard staged. Terminating')
             raise AssertionError('len(samples_to_deliver) != len(hard_staged_samples): {} != {}'.format(len(samples_to_deliver),
                                                                                                         len(hard_staged_samples)))
+
+        hard_staged_misc = []
+        for itm in misc_to_deliver:
+            src_misc = os.path.join(soft_stagepath, itm)
+            dst_misc = os.path.join(hard_stagepath, itm)
+            try:
+                if os.path.isdir(src_misc):
+                    shutil.copytree(src_misc, dst_misc)
+                else:
+                    shutil.copy(src_misc, dst_misc)
+            except Exception, e:
+                logger.error('Miscellaneous file {} has not been hard staged for project {}. Error says: {}'.format(itm, proj, e))
+                logger.exception(e)
+                exit(1)
```

Comment:
if possible, perhaps propagating the exception upwards rather than exiting down here would be preferable?

Reply:
@b97pla : You mean to use `raise` rather then `exit` ?

Reply:
yes; if you raise the exception, the calling method will have to handle it. After all, that's the method that requested the delivery so it should know what to do if the delivery failed (perhaps some cleaning up, setting status in Charon etc.)

Reply:
Makes sense @b97pla :) I just tried to keep in sync with older codes, so I will also change those then 👍